html/modules/custom/reliefweb_fields/src/Plugin/Field/FieldWidget/ReliefWebOptions.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Drupal\reliefweb_fields\Plugin\Field\FieldWidget;
+
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\Plugin\Field\FieldWidget\OptionsButtonsWidget;
+use Drupal\Core\Field\Plugin\Field\FieldWidget\OptionsWidgetBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\reliefweb_utility\Traits\EntityDatabaseInfoTrait;
+
+/**
+ * Plugin implementation of the 'reliefweb_options' widget.
+ *
+ * @FieldWidget(
+ *   id = "reliefweb_options",
+ *   module = "reliefweb_fields",
+ *   label = @Translation("ReliefWeb Options with description"),
+ *   multiple_values = true,
+ *   field_types = {
+ *     "list_integer",
+ *     "list_string",
+ *     "list_float",
+ *     "boolean",
+ *     "entity_reference",
+ *   }
+ * )
+ */
+class ReliefWebOptions extends OptionsButtonsWidget {
+
+  use EntityDatabaseInfoTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
+    $element = OptionsWidgetBase::formElement($items, $delta, $element, $form, $form_state);
+
+    $options = $this->getOptions($items->getEntity());
+    $selected = $this->getSelectedOptions($items);
+
+    // Load all terms, add description.
+    $entity_type_id = $this->getReferencedEntityTypeId();
```

Comment:
The code below seems to only work for the taxonomy terms so maybe it can bail out earlier with something like:

```php
if ($entity_type_id !== 'taxonomy_term') {
   return parent::formElement($items, $delta, $element, $form, $form_state);
}
```